### ISSUE TITLE
Loss (loss_rpn_box_reg) is nan from RPN

### ISSUE THREAD

**Issue by prajjwal1** [pytorch/vision#1120]
My RPN is defined as:
`
rpn = rpn.RegionProposalNetwork(rpn_anchor_generator, rpn_head,rpn_fg_iou_thresh, rpn_bg_iou_thresh,
rpn_batch_size_per_image, rpn_positive_fraction,rpn_pre_nms_top_n, rpn_post_nms_top_n, rpn_nms_thresh,training)
rpn.cuda()
`

Here's my loss:
proposals, proposal_losses = rpn(src_images,src_features,src_targets)

proposal_losses gives: 
`{'loss_objectness': tensor(0.6926, device='cuda:2', grad_fn=<BinaryCrossEntropyWithLogitsBackward>),
 'loss_rpn_box_reg': tensor(nan, device='cuda:2', grad_fn=<DivBackward0>)}
`
I'm getting the backbone features from Resnet (as in docs) which in turn gives an OrderedDict. Image is an object from ImageList. 

My targets are :
`
[{'boxes': tensor([[1324.5386,  275.7656,  699.6948,   19.5117],
          [1324.5386,  275.1152,  702.2983,   19.5117],
          [1307.6157,  184.0605, 1000.4009,   22.7637],
          [1310.8701,  133.9805,  345.6167,   26.0156]], device='cuda:2'),
  'labels': tensor([0., 0., 5., 5.], device='cuda:2'),
  'image_id': tensor([438.], device='cuda:2'),
  'area': tensor([378240., 375708., 117056., 246178.], device='cuda:2'),
  'iscrowd': tensor([0, 0, 0, 0], device='cuda:2')}]
`

I've tried many ways to prevent nan (normalizing images, low lr) , but it still persists. I've tried with two different datasets. Both of them give nan loss in first epoch.  Only 'loss_rpn_box_reg' becomes nan.  What can be the reason ? I also discussed this issue [here](https://discuss.pytorch.org/t/loss-rpn-box-reg-is-nan-torchvision-0-3/49916). [Another thread](https://discuss.pytorch.org/t/rpn-box-reg-loss-is-nan/50480/9).

**Comment by fmassa** [comment]
@prajjwal1 in my case (and I think it might be the same in your case), the problem for NaN in the RPN was due to degenerate boxes.

For example, I would have a box where the width would be 0 or negative (i.e., x0 > x1). So I remove all degenerate boxes from COCO in https://github.com/pytorch/vision/blob/d84fee6d2a64b1a253de89f40e0d3995e6b83578/references/detection/coco_utils.py#L82

This seems to be the case for you, as in your first box, you have
```python
[1324.5386, 275.7656, 699.6948, 19.5117]
```
which seems to indicate that you might be using the boxes in `x0,y0,w,h` format, instead of returning it as `x0,y0,x1,y1` format.

Note that the boxes that the algorithm expects are `x0, y0, x1, y1`.

I'm closing this issue as this seems to be user-error on the box format, but let me know if you have further questions.

**Comment by IdoZada** [comment]
![image](https://user-images.githubusercontent.com/48810056/88067697-fcbccd80-cb77-11ea-89e0-214f5bc260f9.png)
all the values in output dictionary from the model presents nan .
Why its happen?

**Comment by AkshayLaddha943** [comment]
Hi, I encountered the nan issue with my faster_rcnn_resnet50 model, turns out I was using Adam optimizer which led to the values going to Nan, I changed it back to SGD with momentum and weight decay (to which the original architecture was trained) and the results were better, Hope this helps
